govc vm.clone crash

I get a crash when trying to clone a VM:

```
$ govc vm.clone -ds=NetApp-1 OSX_10_11
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4de75f]

goroutine 1 [running]:
panic(0xb05020, 0xc4200100f0)
        /home/gopher/go/src/runtime/panic.go:500 +0x1a0
github.com/vmware/govmomi/govc/vm.(*clone).cloneVM(0xc42005eb60, 0x11b0bc0, 0xc4200102e8, 0x0, 0x1, 0x5)
        /home/gopher/src/github.com/vmware/govmomi/govc/vm/clone.go:249 +0xbf
github.com/vmware/govmomi/govc/vm.(*clone).Run(0xc42005eb60, 0x11b0bc0, 0xc4200102e0, 0xc42005c7e0, 0x0, 0x0)
        /home/gopher/src/github.com/vmware/govmomi/govc/vm/clone.go:194 +0x361
github.com/vmware/govmomi/govc/cli.Run(0xc42000c190, 0x3, 0x3, 0xc4200001a0)
        /home/gopher/src/github.com/vmware/govmomi/govc/cli/command.go:138 +0x484
main.main()
        /home/gopher/src/github.com/vmware/govmomi/govc/main.go:67 +0x66
```

It's crashing here:

```go
func (cmd *clone) cloneVM(ctx context.Context) (*object.Task, error) {

        // search for the first network card of the source
        devices, err := cmd.VirtualMachine.Device(ctx)  // <-----
```

And indeed, `cmd.VirtualMachine` is nil.

It seems that the lines right before the `cloneVM` call,

```go
        if cmd.VirtualMachine, err = cmd.VirtualMachineFlag.VirtualMachine(); err != nil {
                return err
        }
```

Returns `(nil, nil)`.

This is probably my fault, since I'm just starting to tinker with this, but the docs aren't particularly illustrative in how this is supposed to be used.

In particular, it doesn't even say that it accepts an argument:

```
$ govc vm.clone
Usage: govc vm.clone [OPTIONS]

Options:
  -c=0                          Number of CPUs
  -cert=                        Certificate [GOVC_CERTIFICATE]
  -customization=               Customization Specification Name
  -datastore-cluster=           Datastore cluster [GOVC_DATASTORE_CLUSTER]
  -dc=                          Datacenter [GOVC_DATACENTER]
  -debug=false                  Store debug logs [GOVC_DEBUG]
  -ds=                          Datastore [GOVC_DATASTORE]
  -dump=false                   Enable output dump
  -folder=                      Folder [GOVC_FOLDER]
  -force=false                  Create VM if vmx already exists
  -host=                        Host system [GOVC_HOST]
  -host.dns=                    Find host by FQDN
  -host.ip=                     Find host by IP address
  -host.ipath=                  Find host by inventory path
  -host.uuid=                   Find host by UUID
  -json=false                   Enable JSON output
  -k=true                       Skip verification of server certificate [GOVC_INSECURE]
  -key=                         Private key [GOVC_PRIVATE_KEY]
  -m=0                          Size in MB of memory
  -net=                         Network [GOVC_NETWORK]
  -net.adapter=e1000            Network adapter type
  -net.address=                 Network hardware address
  -on=true                      Power on VM
  -persist-session=true         Persist session to disk [GOVC_PERSIST_SESSION]
  -pool=                        Resource pool [GOVC_RESOURCE_POOL]
  -template=false               Create a Template
  -u=https://@10.88.203.10/sdk  ESX or vCenter URL [GOVC_URL]
  -vim-namespace=urn:vim25      Vim namespace [GOVC_VIM_NAMESPACE]
  -vim-version=6.0              Vim version [GOVC_VIM_VERSION]
  -vm=                          Virtual machine [GOVC_VM]
  -waitip=false                 Wait for VM to acquire IP address
```

I see `Usage: govc vm.clone [OPTIONS]` but no `<VM-NAME>` or whatever.

But the code definitely wants a VM name:

```go
func (cmd *clone) Run(ctx context.Context, f *flag.FlagSet) error {
        var err error

        if len(f.Args()) != 1 {
                return flag.ErrHelp
        }

        cmd.name = f.Arg(0)
```

--- a/govc/test/vm.bats
+++ b/govc/test/vm.bats
@@ -435,3 +435,9 @@ load test_helper
   assert_line "Memory: 1024MB"
   assert_line "CPU: 2 vCPU(s)"
 }
+
+@test "vm.clone usage" {
+  # validate we require -vm flag
+  run govc vm.clone enoent
+  assert_failure
+}

--- a/govc/vm/clone.go
+++ b/govc/vm/clone.go
@@ -101,6 +101,17 @@ func (cmd *clone) Register(ctx context.Context, f *flag.FlagSet) {
 	f.BoolVar(&cmd.waitForIP, "waitip", false, "Wait for VM to acquire IP address")
 }
 
+func (cmd *clone) Usage() string {
+	return "NAME"
+}
+
+func (cmd *clone) Description() string {
+	return `Clone VM to NAME.
+Example:
+govc vm.clone -vm template-vm new-vm
+`
+}
+
 func (cmd *clone) Process(ctx context.Context) error {
 	if err := cmd.ClientFlag.Process(ctx); err != nil {
 		return err
@@ -189,6 +200,10 @@ func (cmd *clone) Run(ctx context.Context, f *flag.FlagSet) error {
 
 	if cmd.VirtualMachine, err = cmd.VirtualMachineFlag.VirtualMachine(); err != nil {
 		return err
+	}
+
+	if cmd.VirtualMachine == nil {
+		return flag.ErrHelp
 	}
 
 	task, err := cmd.cloneVM(ctx)

